cmd/remote_client.go

```diff
@@ -262,21 +264,95 @@ func (cli *Client) RemoteLogin(c *clipkg.Context) error {
 }
 
+func extractContractAddressFromAssignmentSyntax(flag string) (string, error) {
+	equalSplit := strings.Split(flag, "=")
+	if len(equalSplit) != 2 {
+		return "", errors.New("too many '=' in" + flag)
+	}
+	return equalSplit[1], nil
+}
+
+// extractEthereumAddressFromWithdrawalCommandline takes the arg containing the
+// specific-contract-address flag, and any following args, and returns the
+// ethereum address as a string, if present, or the empty string if not.
+// Assumes the contract-address flag comes at the end...
+func extractEthereumAddressFromWithdrawalCommandline(args []string) (string, error) {
+	flag := args[0]
+	if strings.Contains(flag, "=") {
+		if len(args) > 1 {
+			return "", fmt.Errorf("too many args for %s", flag)
+		}
+		return extractContractAddressFromAssignmentSyntax(flag)
+	}
+	if len(args) != 2 {
+		return "", fmt.Errorf("wrong number of args for %s", flag)
+	}
+	return args[1], nil
+}
+
+// ParseWithdrawalContractAddress is exposed only for testing purposes. Do not
+// use externally. Checks for valid specification of correctly-capitalized
+// contract address. If it finds one, it returns that. Otherwise, it returns the
+// default ("empty") Address, and any errors it encountered in the flag or
+// during parsing.
+//
+// TODO: go's flags library has a lot of useful parsing tools. Can we use that?
+func ParseWithdrawalContractAddress(args []string) (common.Address, error) {
+	specifiesContractAddress :=
+		(len(args) >= 3) && (!strings.HasPrefix(args[2], "-from-oracle-contract-address") ||
+			!strings.HasPrefix(args[2], "--from-oracle-contract-address"))
+	if !specifiesContractAddress {
+		emptyAddressImplyingDefaultContractAddress := common.Address{}
+		return emptyAddressImplyingDefaultContractAddress, nil
+	}
+	contractAddressUnparsed, err := extractEthereumAddressFromWithdrawalCommandline(
+		args[2:])
+	if err != nil {
+		return utils.ErrorWithEmptyAddress(
+			"could not parse contract address from %v", nil, args[2:])
+	}
+	contractAddress, err := utils.ParseEthereumAddress(contractAddressUnparsed)
+	if err != nil {
+		return utils.ErrorWithEmptyAddress("while parsing %v", err, args[2:])
+	}
+	return contractAddress, nil
+}
+
 // Withdraw will withdraw LINK to an address authorized by the node
 func (cli *Client) Withdraw(c *clipkg.Context) error {
-	if len(c.Args()) < 2 {
+	args := c.Args()
+	if len(args) < 2 {
 		return cli.errorOut(errors.New("withdrawal requires an address and amount"))
 	}
 
-	i, err := strconv.ParseInt(c.Args().Get(1), 10, 64)
+	if len(args) > 4 {
+		return cli.errorOut(errors.New("too many arguments to withdraw"))
+	}
+
+	// XXX: This form of CL-parsing is cumbersome... Why not use go's
+	// parsing mechanism?
+	linkAmount, err := strconv.ParseInt(args.Get(1), 10, 64)
 
 	if err != nil {
-		return err
+		return cli.errorOut(multierr.Combine(
+			errors.New("while parsing LINK withdrawal amount"), err))
+	}
+
+	contractAddress, cErr := ParseWithdrawalContractAddress(args)
```

Comment:
✂️ 
I would like to use https://github.com/urfave/cli#flags and cut this alongside their supporting functions (`extractEthereumAddressFromWithdrawalCommandline`).

An example of this in our code base can be found here: https://github.com/smartcontractkit/chainlink/blob/master/main.go#L69-L74

```golang
{
  Name:   "login",
  Usage:  "Login to remote client by creating a session cookie",
  Action: client.RemoteLogin,
  Flags: []cli.Flag{
    cli.StringFlag{ // <----- String flag!
      Name:  "file, f",
      Usage: "text file holding the API email and password needed to create a session cookie",
    },
  },
},
...

func (cli *Client) RemoteLogin(c *clipkg.Context) error {
	sessionRequest, err := cli.buildSessionRequest(c.String("file")) <--- Usage
```

Reply:
Thanks, adjusted.